crates/wallet/src/lib.rs

```diff
@@ -386,4 +386,45 @@ mod tests {
             );
         }
     }
+
+    mod validation {
```

Comment:
nit: we should get rid of nested mods in tests mod, this is just bloat -.-

Reply:
imo it makes test output way more readable, but can remove

Reply:
removed